sources/cfsignal/src/cloudformation.rs

```diff
@@ -0,0 +1,55 @@
+use crate::error::{self, Result};
+use imdsclient::ImdsClient;
+use log::info;
+use rusoto_cloudformation::{CloudFormation, CloudFormationClient, SignalResourceInput};
+use rusoto_core::Region;
+use snafu::{OptionExt, ResultExt};
+use std::str::FromStr;
+
+pub async fn signal_resource(
+    stack_name: String,
+    logical_resource_id: String,
+    status: String,
+) -> Result<()> {
+    info!("Connecting to IMDS");
+    let mut client = ImdsClient::new();
+    let instance_id = get_instance_id(&mut client).await.unwrap();
+    let region = get_region(&mut client).await.unwrap();
+
+    info!(
+        "Region: {:?} - InstanceID: {:?} - Signal: {:?}",
+        region, instance_id, status
+    );
+
+    let client = CloudFormationClient::new(Region::from_str(&region).unwrap());
```

Comment:
We need a new error variant added to `cfsignal/src/error.rs`:
```
    #[snafu(display("Unable to parse '{}' as a region: {}", region, source))]
    RegionParse {
        region: String,
        source: ParseRegionError,
    },
```
Then
```suggestion
    let client = CloudFormationClient::new(Region::from_str(&region).context(error::RegionParse { region })?);
```